src/Application/Commands/SendStatistics.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MatchBot\Application\Commands;
+
+use Aws\CloudWatch\CloudWatchClient;
+use MatchBot\Domain\DonationRepository;
+use Symfony\Component\Console\Attribute\AsCommand;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * Send CloudWatch Metrics values of:
+ * * overall completion rate of donations from ~16 minutes old thru ~1 minutes old, if 20+
+ *   such donations started in that time
+ * * donations created (in previous full minute before task start)
+ * * donations collected (in previous full minute before task start)
+ */
+#[AsCommand(
+    name: 'matchbot:send-statistics',
+    description: 'Sends CloudWatch headline figures on donation initiation and collection'
+)]
+class SendStatistics extends LockingCommand
+{
+    public function __construct(
+        private CloudWatchClient $cloudWatchClient,
+        private DonationRepository $donationRepository
+    ) {
+        parent::__construct();
+    }
+
+    protected function doExecute(InputInterface $input, OutputInterface $output): int
+    {
+        $time = time();
+        $startOfThisMinute = new \DateTimeImmutable('@' . ($time - ($time % 60)));
+        $completionRate = $this->makeMetric(
+            'CompletionRate',
+            $this->donationRepository->getRecentHighVolumeCompletionRatio($startOfThisMinute),
+            $startOfThisMinute,
+        );
+        $donationsCreated = $this->makeMetric(
+            'DonationsCreated',
+            $this->donationRepository->getDonationsJustCreated($startOfThisMinute),
+            $startOfThisMinute,
+        );
+        $donationsCollected = $this->makeMetric(
+            'DonationsCollected',
+            $this->donationRepository->getDonationsJustCollected($startOfThisMinute),
+            $startOfThisMinute,
+        );
+        $notNullMetrics = [$donationsCreated, $donationsCollected];
+        if (null !== $completionRate['Value']) {
+            $notNullMetrics[] = $completionRate;
+        }
+
+        $this->cloudWatchClient->putMetricData([
+            'Namespace' => 'TbgMatchBot',
+            'MetricData' => $notNullMetrics,
+        ]);
+
+        $count = count($notNullMetrics);
+        $output->writeln("Sent $count metrics to CloudWatch");
+
+        return 0;
+    }
+
+    /**
+     * Value in default 'Unit' mode accepts doubles, so we leave this with the default for both float
+     * & int values.
+     */
+    private function makeMetric(string $name, float | int | null $value, \DateTimeImmutable $timestamp): array
+    {
+        $appEnv = getenv('APP_ENV');
+        \assert(is_string($appEnv)); // crash if somehow missing.
```

Comment:
I think it's worth injecting the appEnv - something like this using the existing `\MatchBot\Application\Environment` enum class. Except I'd put move the strtolower inside that class (or replace it with a match)

```suggestion
        $appEnv = strtolower($this->environment->name);
```

Reply:
Yep makes sense, forgot we already had that. Done in d2d92a285d23985fa621cb56ac63e47a25f3f5b8